torch/_dynamo/variables/nn_module.py

```diff
@@ -801,3 +776,124 @@ def __init__(self, value, **kwargs):
         else:
             # this makes us behave like a usual UnspecializedNNModuleVariable for guarding purposes
             self.source = NotNNModuleSource(source)
+        self.module_key = module_key
+
+    def call_method(
+        self, tx, name, args: List[VariableTracker], kwargs: Dict[str, VariableTracker]
+    ) -> VariableTracker:
+        key = self.module_key
+        options = VariableTracker.propagate(self, args, kwargs.values())
+
+        named_embed = functools.partial(
+            _named_embed,
+            tx=tx,
+            key=key,
+            source_cls=FSDPNNModuleSource,
+            source=self.source,
+            options=options,
+        )
+        wrap_values = functools.partial(
+            _wrap_values,
+            tx=tx,
+            key=key,
+            source_cls=FSDPNNModuleSource,
+            source=self.source,
+            options=options,
+        )
+        get_kwargs = functools.partial(
+            _get_kwargs, mod=self.value, name=name, args=args, kwargs=kwargs
+        )
+
+        if name == "buffers":
+            return wrap_values(self.value.named_buffers(**get_kwargs("recurse")))
+        elif name == "named_buffers":
+            result = []
+            for name, buffer in self.value.named_buffers(
+                **get_kwargs("prefix", "recurse", "remove_duplicate")
+            ):
+                result.append(named_embed(name, buffer))
+            return variables.ListIteratorVariable(
+                result, mutable_local=MutableLocal(), **options
+            )
+        elif name == "children":
+            assert not (args or kwargs)
+            return wrap_values(self.value.named_children())
+        return super().call_method(tx, name, args, kwargs)
+
+    def var_getattr(self, tx, name):
+        if name in ["named_buffers", "children", "buffers"]:
+            # Route this to produce a ListIteratorVariable instead of getting the generator
+            return variables.LambdaVariable(
+                lambda *args, **kwargs: self.call_method(tx, name, args, kwargs)
+            ).add_options(self)
+        return super().var_getattr(tx, name)
+
+    def as_python_constant(self):
```

Comment:
should this be at the UnspecNN layer?

Reply:
(actually, for most of these things- buffers, children, etc- should these be supported at the unspec layer ideally? (i see the source changes, so yea there is a difference, i'm ok leaving it this way just curious

Reply:
Hmm, I didn't want to blanket change all unspec. That's a huge change - I suppose I could revisit this after and see if we can collapse it down a layer.